SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialSender.cpp

```diff
@@ -541,36 +541,82 @@ void USpatialSender::SendRPC(TSharedRef<FPendingRPCParams> Params)
 		break;
 	}
 	case SCHEMA_NetMulticastRPC:
-	case SCHEMA_ClientUnreliableRPC:
-	case SCHEMA_ServerUnreliableRPC:
 	{
-		Worker_ComponentId ComponentId;
-		if (RPCInfo->Type == SCHEMA_ClientUnreliableRPC)
+		FUnrealObjectRef TargetObjectRef(PackageMap->GetUnrealObjectRefFromNetGUID(PackageMap->GetNetGUIDFromObject(TargetObject)));
+		if (TargetObjectRef != FUnrealObjectRef::UNRESOLVED_OBJECT_REF)
 		{
-			ComponentId = SpatialConstants::SERVER_RPC_ENDPOINT_COMPONENT_ID;
-		}
-		else if (RPCInfo->Type == SCHEMA_ServerUnreliableRPC)
-		{
-			ComponentId = SpatialConstants::CLIENT_RPC_ENDPOINT_COMPONENT_ID;
+			EntityId = TargetObjectRef.Entity;
+			Worker_ComponentId ComponentId = SpatialConstants::NETMULTICAST_RPCS_COMPONENT_ID;
+
+			Worker_ComponentUpdate ComponentUpdate = CreateRPCEventUpdate(TargetObject, Params->Function, Params->Parameters.GetData(), ComponentId, RPCInfo->Index, UnresolvedObject);
+
+			if (!UnresolvedObject)
```

Comment:
I think this function needs a refactor. It's really long and nested. For now we can probably turn these kinds of checks around and break early if an object is unresolved.

Reply:
I agree. As discussed, we can also make this function queue the RPCs and send them later, rather than sending them at once. That will allow some simplifications as well as help preserve the ordering for cases with unresolved objects.

Reply:
I've simplified this function. However, I changed my mind about queueing - I think it would be unnecessary overhead to queue when the RPC can be sent straightaway.

Reply:
Now the RPCs are queued, resolving this comment.